RentACar/RentACar/CRUD/CarCRUD.cs

```diff
@@ -1,69 +1,57 @@
-﻿using System;
-using System.IO;
-using System.Text.Json;
+﻿using System.IO;
+using RentACar.Utilities;
 using System.Collections.Generic;
 using System.Linq;
-
+using RentACar.CRUD;
 
 namespace RentACar
 {
-    public class CarCRUD
+    public class CarCRUD : ICarCRUD
     {
-        private readonly string _jsonFile = @"..\..\..\CRUD\Cars.json";
-
-        private List<Car> DeserializeJsonToList(string jsonFile)
-        {
-            string existingJsonToString = File.ReadAllText(jsonFile);
-            List<Car> deserializedJson = JsonSerializer.Deserialize<List<Car>>(existingJsonToString);
-
-            return deserializedJson;
-        }
-
-        private void SerializeListToJson(List<Car> cars)
-        {
-            var options = new JsonSerializerOptions { WriteIndented = true };
-
-            string listToString = JsonSerializer.Serialize(cars, options);
-            File.WriteAllText(_jsonFile, listToString);
-        }
+        private readonly string _jsonFile = @"..\..\..\Data\Cars.json";
 
         public Car Create(Car car)
         {
-            var cars = new List<Car>();
+            var cars = (File.Exists(_jsonFile)) ? CrudHelper.DeserializeJsonToList(_jsonFile) : new List<Car>();
 
-            if (File.Exists(_jsonFile))
-            {
-                cars = DeserializeJsonToList(_jsonFile);
+            car.Id = CrudHelper.GetNewId(cars);
+            cars.Add(car);
 
-                car.Id = cars.Last().Id + 1;
-                cars.Add(car);
-            }
-            else
-            {
-                car.Id = 0;
-                cars.Add(car);
-            }
-
-            SerializeListToJson(cars);
+            CrudHelper.SerializeListToJson(cars, _jsonFile);
 
             return car;
         }
 
         public Car Get(int id)
         {
-            // Code here
-            return new Car();
+            if (!File.Exists(_jsonFile)) return null;
+
+            var cars = CrudHelper.DeserializeJsonToList(_jsonFile);
+
+            var car = cars.Where(e => e.Id == id).FirstOrDefault();
+
+            return car ?? null;
         }
 
         public Car Update(Car car)
         {
-            // Code here
-            return new Car();
+            if (!File.Exists(_jsonFile)) return null;
```

Comment:
you could move this check to the helper class and perhaps return a new empty car list when the file does not exist? that way you would re-utilize this check everywhere instead of replicating it all over the place

Reply:
You are right! i didn't noticed that.. But instead return an empty car list, Should not i create a .json file with an empty list?

Reply:
yes of course! it's a feature design you have full control over, mine was just a suggestion

Reply:
Perfect! Thanks!!

Reply:
I thought about it better. You are right 😋, because if i do what i said above, i am replicating all over the place the deserialization. I will make the changes in a moment and pushing it.. Thanks again!